### PR TITLE
feat(cmp): add on_config_done callback

### PR DESCRIPTION
**Pull request by mcartaud**
# Description

User can use a `on_config_done` property in nvim-cmp config. It allow user to hook on the plugin and make additionnal configuration after the plugin is setup.

## How has this been tested

1. See that completion shows up when writing a commit for exemple
2. Add 
```lua
lvim.builtin.cmp.on_config_done = function(cmp)
  cmp.setup.filetype("gitcommit", {
    enabled = false,
  })
end
```
In the nvim config file
4. Reload LunarVim
5. Commit something in a repo and see that completion doesn't open

Signed-off-by: Mathieu Cartaud <dev@mcartaud.com>

### CODE REVIEW

**Comment by LostNeophyte** [comment]
The `enabled` function is configurable so it's already possible

**Comment by LostNeophyte** [comment]
The cmp.setup.filetypes may not be possible to set but i think having an on_config_done callback like with other plugins would be a better solution

**Comment by mcartaud** [comment]
> The `enabled` function is configurable so it's already possible

Ok, I though that it would be best to have this case available by configuration instead of by overriding all work done in the `enabled` function, because it might be harder to upgrade for the user if LunarVim code changes

> The cmp.setup.filetypes may not be possible to set but i think having an on_config_done callback like with other plugins would be a better solution

I was not aware of this mechanism, will rework the PR

**Comment by mcartaud** [comment]
This new version adds an `on_config_done` property on the cmp object and removes the configuration to disable cmp in comments

**Comment by LostNeophyte** [comment]
Is it ready for review?

File: lua/lvim/core/cmp.lua [hunk 1]
```diff
@@ -367,6 +368,9 @@ function M.setup()
       })
     end
   end
```
Comment (LostNeophyte) [inline lua/lvim/core/cmp.lua:370]:
```suggestion
  end
  
```

**Comment by mcartaud** [comment]
Thanks for the review

**Comment by LostNeophyte** [comment]
Update the pr description and all looks good to me, i've never seen the checks take so long btw

**Comment by mcartaud** [comment]
> i've never seen the checks take so long btw

Yes, lot's of things to test in this review :)

It's all done, thanks

**Review by LostNeophyte — APPROVED** [review]
LGTM, thanks :)

**Review by kylo252 — APPROVED** [review]
Thanks!

**Pull request merged**